colander/__init__.py

```diff
@@ -2120,6 +2120,8 @@ def clone(self):
         dictionaries are preserved."""
         children = [node.clone() for node in self.children]
         cloned = self.__class__(self.typ, *children)
+        cloned.children = []
+        _add_node_children(cloned, children)
```

Comment:
If we are manually setting `cloned.children` after creating the instance, then we likely shouldn't be passing them into the `self.__class__`.  Also, a comment here on why we don't do that would be a Good Thing.